[MTM-64955] Angular 20 upgrade and Web SDK 1023 upgrade release note and upgrade guide (#3905)
SoftwareAG/c8y-docs: content/web/upgrade-bundle/upgrading-to-angular-20.md

---
title: Upgrading from Angular 19 to Angular 20
layout: redirect
weight: 42
---

Starting with version 1023.0.0, the Web SDK supports Angular 20. The following configuration changes are required before you can run the application:

- Run the command `ng update @angular/core@20 @angular/cli@20` to update Angular core and CLI to version 20.
- Update all `@c8y` dependencies to version `1023.x.x` in your *package.json*.
- Update `ngx-bootstrap` to version `20.0.2`.
- Update `@angular/cdk` to version `20.x.x`.
- Update TypeScript to version `5.9.3` or higher.
- Ensure compatibility for `Node.js`, `TypeScript`, `RxJS`: [Version compatibility](https://angular.dev/reference/versions#actively-supported-versions).
- Follow the `Angular 20` upgrade guide: [Updating to version 20](https://angular.dev/update-guide?v=19.0-20.0&l=2).
- Adjust the `main.ts` and `bootstrap.ts` files according to the git diffs [mentioned below](#track-changes-between-releases).

> ### Breaking changes

Reviewer: ```suggestion
### Breaking changes {#breaking-changes}
```

The Web SDK version 1023.0.0 introduces multiple breaking changes.

#### Global time context API changes

> The API for global time context has been redesigned. Widget configurations now use `widgetInstanceGlobalTimeContext`, `widgetInstanceGlobalAutoRefreshContext`, and related properties instead of previous implementations. Review your widget implementations that interact with global time context and update them accordingly.

Reviewer 2: This is not correct. We need to discuss how to proceed with it.

Reviewer 2: ```suggestion
The global time context integration API has been redesigned. Custom widget implementations may require updates. Further details will be provided in upcoming documentation.
```

@BeateRixen  Is this statement okay with you? Our new feature will receive some changes that will affect its API. As it's still a new feature not yet used by customers, we want to simplify it. However, I'd like to avoid adding details until it's done (I'm working on it right now). The plan would be to update this statement when it's finished.

Reviewer: Yes, sounds ok to me.

#### ngx-translate upgrade

The `@ngx-translate/core` package has been upgraded to version 17.0.0. A separate `TranslateService` instance is now provided per plugin. If your code directly utilizes the `TranslateService`, you may need to adapt your implementation to account for this scoped service approach.

#### QueriesUtil type additions

The `QueriesUtil` class now includes type definitions. Depending on your usage of this class, you may encounter TypeScript compilation errors that need to be resolved by adding appropriate type annotations.

#### BulkSingleOperationsListModule removed

The `BulkSingleOperationsListModule` has been removed. Use `SingleOperationsListComponent` as a standalone component instead.

> ### Track changes between releases

Reviewer: ```suggestion
### Track changes between releases {#track-changes-between-releases}
```
